python/packages/autogen-ext/src/autogen_ext/task/_source_match_termination.py

```diff
@@ -0,0 +1,37 @@
+from typing import Sequence, List
+
+from autogen_agentchat.base import TerminationCondition, TerminatedException
+from autogen_agentchat.messages import StopMessage, AgentMessage
+
+
+class SourceMatchTermination(TerminationCondition):
```

Comment:
Sorry. I initially misunderstood the trigger. Since the trigger is the last source mentioned. I am wondering, if we don't need to limit to the last message's source. Any message that matches the source can trigger termination. The scenario I have in mind is that a summary agent finishes responding then the group chat terminates. 

What's the scenario you have in mind for the trigger?

Reply:
I have the same senario. The only difference is allowing the user to start another conversation with the same chat history.
 So, in my case, I need to check the last message, but not any message

Reply:
The termination condition is applied after the messages have been processed by the agent. So, the chat history is preserved.

Reply:
yeah, but if we have chat history from last conversation and then check any message that matches the source, conversation will alway terminate after 1 agent talking even if it's not the summary agent that we wanted. because the summary agent message existed in history

Reply:
The termination condition is invoked on the delta since the last time it was invoked rather than the whole history. So it won't see the same message twice.

Reply:
Yeah, you're right. Sorry, i missed the senario Nested chat. In that scenario, if we check any message to match the source, then this will allow terminate conversation when specific agents are involved in nested chat. Is that what you mean? I think that makes sense.

I can update this logic on the next commit if you want. Please let me know.

Reply:
Yes. Let's change the logic to terminate whenever an agent source is matched.

> this will allow terminate conversation when specific agents are involved in nested chat. Is that what you mean? 

yes. just to keep it simple, the termination condition in the parent team will be applied on the flattened messages from all agents.

Reply:
sorry for the late, i updated the PR, please check again